.github/workflows/publish-preview.yml

```diff
@@ -0,0 +1,28 @@
+name: Publish a preview build
+
+on: issue_comment
+
+jobs:
+  publish-preview:
+    permissions:
+      contents: write
+    if: ${{ github.event.issue.pull_request && github.event.action == 'created' && startsWith(github.event.comment.body, '@metamaskbot publish-preview') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.sha }}
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: yarn
+          registry-url: 'https://npm.pkg.github.com'
+          scope: '@MetaMask'
+      - run: yarn --immutable
+      - run: yarn prepare-preview-builds
+      - run: yarn build
+      - name: Publish preview build
+        run: yarn publish-previews
+        env:
+          YARN_NPM_AUTH_TOKEN: ${{ secrets.GITHUB_NPM_TOKEN }}
```

Comment:
Ideally we'd comment after this, linking to the packages and giving basic instructions on how to use them

Reply:
Done in https://github.com/MetaMask/controllers/pull/965/commits/e8f70ee06b180c987631529f363718fa128b0d1c

It's a starting point at least